packages/oui-dual-list/src/dual-list.controller.js

```diff
@@ -0,0 +1,251 @@
+import { addBooleanParameter, addDefaultParameter } from "@oui-angular/common/component-utils";
+export default class {
+    constructor ($q, $element, $attrs, ouiDualListProvider) {
+        "ngInject";
+
+        this.$q = $q;
+        this.$element = $element;
+        this.$attrs = $attrs;
+        this.dualListProvider = ouiDualListProvider;
+    }
+
+    $onInit () {
+        addDefaultParameter(this, "sourceListLabel", this.dualListProvider.translations.sourceListLabel);
+        addDefaultParameter(this, "targetListLabel", this.dualListProvider.translations.targetListLabel);
+        addDefaultParameter(this, "moveAllLabel", this.dualListProvider.translations.moveAllLabel);
+        addDefaultParameter(this, "removeAllLabel", this.dualListProvider.translations.removeAllLabel);
+        addDefaultParameter(this, "addLabel", this.dualListProvider.translations.addLabel);
+        addDefaultParameter(this, "sourceListEmptyLabel", this.dualListProvider.translations.sourceListEmptyLabel);
+        addDefaultParameter(this, "targetListEmptyLabel", this.dualListProvider.translations.targetListEmptyLabel);
+        addDefaultParameter(this, "property", null);
+        addBooleanParameter(this, "bulkActionEnabled");
+
+        this.sourceList = this.sourceList || [];
+        this.targetList = this.targetList || [];
+        this.onAdd = this.onAdd || null;
+        this.onRemove = this.onRemove || null;
+
+        this.loadingMap = {};
+        this.sourceListLoading = false;
+        this.targetListLoading = false;
+        this.loadSourceList();
+        this.loadTargetList();
+    }
+
+    $postLink () {
+        this.$element.addClass("oui-dual-list");
+        this.sourceCloseIcon = this.$element[0].querySelector(".oui-dual-list__source > .oui-dual-list__header > .oui-dual-list__header_toggle > .oui-dual-list__header_toggle_up");
```

Comment:
I think we misunderstood ourself for this one :)

Unless if really necessary (eg. `this.$element.addClass("oui-dual-list");`, to add classname on root element), all style/class management must be done in the HTML template using `ngClass`. We must avoid DOM modifications in a jQuery-like way in our AngularJS code. Generally, we tend to do only DOM manipulations in the `$postLink`, wrapped with `$timeout`.

To show/hide objects, you have directives like `ngIf`, `ngHide` or `ngShow`.
In this way, methods like `onTargetContentClose()`, `onTargetContentOpen()`, `onSourceContentClose()` and `onSourceContentOpen()` should be really simplified.

cf. my other comment :)

Reply:
Here I need to add a class to the component root element. If I don't set its display to flex. Scroll does not work. Is there any other way? I have wrapped this code in setTimeout.